homeassistant/components/modbus/binary_sensor.py

```diff
@@ -126,7 +128,7 @@ def __init__(
     async def async_added_to_hass(self):
         """Handle entity which will be added."""
         async_track_time_interval(
-            self.hass, lambda arg: self.update(), self._scan_interval
+            self._hass, lambda arg: self.update(), self._scan_interval
```

Comment:
This isn't needed. `hass` is set on the entity when it is added to Home Assistant.

https://github.com/home-assistant/core/blob/98edd58c558696a4cb38780446dac9ff81124214/homeassistant/helpers/entity.py#L568-L581

Reply:
Thanks for explaining I got a bit scared